modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -1025,6 +1064,64 @@ impl LightningClientModule {
         Ok(lnurl)
     }
 
+    /// Subscribe to a stream of Lightning events (both send and receive).
+    ///
+    /// Emits an event each time a send or receive operation reaches a
+    /// significant state:
+    /// - [`SendPaymentEvent`]: When a send transaction is successfully funded
+    /// - [`SendPaymentUpdateEvent`]: When a send operation reaches a final
+    ///   state (success, refunded, or failure)
+    /// - [`ReceivePaymentEvent`]: When a receive operation completes and
+    ///   transitions to claiming
+    ///
+    /// The stream efficiently waits for new events using the notification
+    /// system instead of polling, filtering for only this module's
+    /// Lightning payment events.
+    pub fn subscribe_payment_events(
+        &self,
+    ) -> impl Stream<Item = LightningPaymentEventWithTimestamp> {
+        self.client_ctx
+            .subscribe_persisted_events()
+            .filter_map(move |entry| async move {
```

Comment:
Does this go through all the events all the time? Too inefficient for long-running clients.